bazel/src/main/java/com/google/api/codegen/bazel/BazelBuildFileView.java

```diff
@@ -281,17 +336,24 @@ private Set<String> mapJavaGapicDeps(Set<String> protoImports) {
     return javaImports;
   }
 
-  private Set<String> mapJavaGapicTestDeps(Set<String> protoImports) {
+  private Set<String> mapJavaGapicTestDeps(Set<String> protoImports, String transport, String name) {
     Set<String> javaImports = new TreeSet<>();
+    if (transport.contains("grpc")) {
+      javaImports.add(String.format(":%s_java_grpc", name));
+    }
+
     for (String protoImport : protoImports) {
-      if (protoImport.endsWith(":iam_policy_proto")
-          || protoImport.endsWith(":policy_proto")
-          || protoImport.endsWith(":options_proto")) {
+      boolean iamDep = protoImport.endsWith(":iam_policy_proto")
+        || protoImport.endsWith(":policy_proto")
+        || protoImport.endsWith(":options_proto");
+
+      if (iamDep && transport.contains("grpc")) {
         javaImports.add(replaceLabelName(protoImport, ":iam_java_grpc"));
-      } else if (protoImport.endsWith(":location_proto")) {
+      } else if (protoImport.endsWith(":location_proto") && transport.contains("grpc")) {
```

Comment:
nit: maybe factor out the gRPC check to an outer `if` statement? I think it would make the logic clearer.

Reply:
If consolidated this into a variable and used that everywhere, thanks!